examples/bridge-app/linux/DynamicDevice.cpp

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    Copyright (c) 2019 Google LLC.
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DynamicDevice.h"
+
+void DynamicAttributeList::Add(chip::AttributeId id, EmberAfAttributeType type, uint16_t maxSize, EmberAfClusterMask mask)
+{
+    // Do what DECLARE_DYNAMIC_ATTRIBUTE does.
+    mAttributeDecls.emplace_back(EmberAfAttributeMetadata{
+        id, type, maxSize, (uint8_t) (mask | ZAP_ATTRIBUTE_MASK(EXTERNAL_STORAGE)), ZAP_EMPTY_DEFAULT() });
```

Comment:
I don't think a nonzero maxsize is tested with the "external" mask...  Things may or may not work right.

Reply:
Yes, I have since audited the relevant code and there was one bug but it wouldn't bite you unless you use the SINGLETON flag.